src/context/mod.rs

```diff
@@ -425,13 +425,16 @@ impl Context {
                 raw::reply_with_big_number(self.ctx, s.as_ptr().cast::<c_char>(), s.len())
             }
 
-            Ok(RedisValue::VerbatimString((format, mut data))) => {
-                let mut final_data = format.as_bytes().to_vec();
-                final_data.append(&mut data);
+            Ok(RedisValue::VerbatimString((mut format, data))) => {
+                while format.len() < 3 {
```

Comment:
But then it will duplicate the string no?

Reply:
The assumption here is that 99% of the time this will not be needed, but thinking about it now, if the format must be a 3 chars long then maybe we should not hold a string there at all but a slice of len 3?

Reply:
Yes